client/vat/form-modal/__tests__/index.tsx

```diff
@@ -58,31 +57,4 @@ describe( 'Tax details modal', () => {
 			screen.getByRole( 'dialog', { name: 'Set your tax details' } )
 		).toMatchSnapshot();
 	} );
-
-	it( 'should close when clicking on the dismiss button', () => {
```

Comment:
Removed because this test was not kosher. It was attempting to manually re-render a component as if the component itself was responsible for re-rendering on state change.
The same assertions are present above.
The test should have had an internal `useState`, to be correctly implemented. But at that point, we're not testing the component anymore, we're testing some logic that doesn't exist in the component itself.